src/scripts/pokemons/PokemonFactory.ts

```diff
@@ -35,9 +35,9 @@ class PokemonFactory {
         const catchRate: number = this.catchRateHelper(basePokemon.catchRate);
         const exp: number = basePokemon.exp;
         const level: number = this.routeLevel(route, region);
-        const heldItem: BagItem = this.generateHeldItem(basePokemon.heldItem, GameConstants.ROUTE_HELD_ITEM_MODIFIER);
         const money: number = this.routeMoney(route,region);
         const shiny: boolean = this.generateShiny(GameConstants.SHINY_CHANCE_BATTLE);
+        const heldItem: BagItem = this.generateHeldItem(basePokemon.heldItem, GameConstants.ROUTE_HELD_ITEM_MODIFIER, shiny);
```

Comment:
Instead of modifying `generateHeldItem` you can just do:
```suggestion
        const heldItem: BagItem = shiny ? basePokemon.heldItem : this.generateHeldItem(basePokemon.heldItem, GameConstants.ROUTE_HELD_ITEM_MODIFIER);
```

Reply:
Doing so would bypass the item requirement.

Reply:
Right, requirements, ignore me